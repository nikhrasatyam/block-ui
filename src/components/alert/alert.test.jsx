import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import Enzyme from 'enzyme';
import UiStore from '../../stores/ui.store';


import Alert from './alert';

describe('Alert Test Cases', () => {
  let uiStore;
  beforeEach(async () => {
    uiStore = new UiStore();
  });
  afterEach(() => { uiStore = null; });

  test('should render alert component when uiStore.alert.message is empty', () => {
    const tree = renderer.create(<Provider uiStore={uiStore}><Alert /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render alert component when uiStore.alert is defined', () => {
    uiStore.setAlert({ type: 'success', title: 'success', message: 'test' });
    const tree = renderer.create(<Provider uiStore={uiStore}><Alert /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have message component when uiStore.alert.message is defined ', async () => {
    uiStore.setAlert({ type: 'success', title: 'success', message: 'test' });
    expect(uiStore.alert.message).toBe('test');
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore}><Alert /></Provider>);
    expect(wrapper.find('Message')).toHaveLength(1);
  });

  test('should not have message component when uiStore.alert.message is empty ', async () => {
    expect(uiStore.alert.message).toBe('');
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore}><Alert /></Provider>);
    expect(wrapper.find('Message')).toHaveLength(0);
  });
});
