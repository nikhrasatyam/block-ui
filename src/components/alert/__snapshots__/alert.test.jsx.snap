// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Test Cases should render alert component when uiStore.alert is defined 1`] = `
<div
  className="ui small positive message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="header"
  >
    success
  </div>
  <p>
    test
  </p>
</div>
`;

exports[`Alert Test Cases should render alert component when uiStore.alert is undefined 1`] = `null`;
